ElectronClient/app/gui/NotebookPropertiesDialog.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import { useState, useEffect } from 'react';
+
+const { _ } = require('lib/locale.js');
+const { themeStyle, buildStyle } = require('../theme.js');
+const DialogButtonRow = require('./DialogButtonRow.min');
+
+const Folder = require('lib/models/Folder.js');
+
+interface FolderPropertiesDialogProps {
+	theme: number,
+	folderId: string,
+	onClose: Function,
+}
+
+function styles_(props:FolderPropertiesDialogProps) {
+	return buildStyle('FolderPropertiesDialog', props.theme, (theme: any) => {
+		return {
+			folderLabel: {
+				...theme.textStyle,
+				flex: 1,
+				display: 'flex',
+				fontWeight: 'bold',
+				color: theme.color,
+			},
+			folderSaveButton: {
+				background: 'none',
+				border: 'none',
+			},
+			folderSaveButtonIcon: {
+				color: theme.color,
+				fontSize: '1.4em',
+			},
+			titleForm: {
+				display: 'inline-block',
+				color: theme.color,
+				backgroundColor: theme.backgroundColor,
+				border: '1px solid',
+				borderColor: theme.dividerColor,
+				marginLeft: '3em',
+				width: '6em',
+			},
+			iconForm: {
+				display: 'inline-block',
+				color: theme.color,
+				backgroundColor: theme.backgroundColor,
+				border: '1px solid',
+				borderColor: theme.dividerColor,
+				marginLeft: '3.6em',
+				width: '6em',
+			},
```

Comment:
For all these styles, is it not possible to re-use existing styles from the NotePropertiesDialog for example? It's good that you made it based on the theme but there's still a lot of custom styling and we need the app to have a uniform styling.

Reply:
I'm going to be working a good part of the day, so i won't be at my computer for a bit, but i do have some ideas about how to improve this. I would definitely need to do some testing. 

Also, since Joplin supports a wide range of languages and their respective typesets, is there logic somewhere for finding out how wide a space needs to be for specific strings in one of the wider typesets? I don't know if this question makes sense or not. I want to make sure that where the names of each property will look great no matter what language it's in.

Reply:
Don't worry about the styling for now, let's try to get the rest of it working first. For styling, it might have to be a wider scale refactoring.